### DISCUSSION TITLE
Merge approved MRs/PRs

### DISCUSSION THREAD

**Discussion by mikestreety** [renovatebot/renovate#25096] [Suggest an Idea]
### Tell us more.

We currently have Renovate to auto-merge patches but require manual intervention for minor updates. We use Gitlab

What would be great is if you could enable Renovate to auto-merge any MRs which have been approved in Gitlab.

So the workflow would be

- Renovate runs, makes an MR for a minor update
- Developer reviews the change and approves the MR
- Renovate runs, see's MR is approved and merges it.

Is this possible or is it a pipedream?

**Comment by rarkins** [comment]
I don't see any reason why it's impossible, but it's definitely not supported today. I guess you're after a setting like "automerge after approval". Just remember that it might be some time (e.g. even hours) before Renovate runs again on that same repo after approval, also if it's Renovate's automerge (instead of GitLab automerge) then it will merge at most one MR per run

**Reply by mikestreety** [reply]
That sounds exactly what I want and the outcome is what I'm after too. We potentially have permissions where someone can approve but not merge, so this would help with that

**Reply by rarkins** [reply]
It might work to use automergeType=after-pr-approval (https://docs.renovatebot.com/configuration-options/#automergetype). PR welcome if you wish

**Reply by mikestreety** [reply]
Thanks, i'll have a look. Shall I make an issue for this?

**Reply by rarkins** [reply]
#25118
